packages/vitest/src/node/reporters/json.ts

```diff
@@ -47,33 +71,80 @@ export class JsonReporter implements Reporter {
   protected async logTasks(files: File[]) {
     const suites = getSuites(files)
     const numTotalTestSuites = suites.length
-    const tests = getTests(files)
-    const numTotalTests = tests.length
+    const fileToTestCases = new Map<File, Test[]>()
 
+    for (const file of files) {
+      const relatedTests = getTests([file])
+      fileToTestCases.set(file, relatedTests)
+    }
+
+    const tests = [...fileToTestCases.values()].flat()
```

Comment:
I think there is no problem in calling `getTests` twice. This line could remain as `const tests = getTests(files)` and the `fileToTestCases` map isn't needed then. You can directly use `for (const file of files) {` in line 94.